.travis.yml

```diff
@@ -31,16 +31,23 @@ jobs:
   include:
     - name: lint-checks
       stage: test
+      compiler: clang
       addons:
         apt:
           packages: ['python3-pip']
       install: pip3 install --user flake8==3.7.8
       script:
         - flake8
-        - ./clang-tidy-diff.sh
         - ./clang-format-diff.sh
         # ensure generated parser is up to date
         - ./scripts/gen-s-parser.py | diff src/gen-s-parser.inc -
+        - cmake ${TRAVIS_BUILD_DIR} -G Ninja
+            -DCMAKE_C_FLAGS="$COMPILER_FLAGS"
+            -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS"
+            -DCMAKE_EXE_LINKER_FLAGS="$LINKER_FLAGS"
+            -DCMAKE_INSTALL_PREFIX=install
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

Comment:
Oh sorry, I didn't realize clang-tidy depending on building the project first.  

I move these sanitizers to their own step, separate from the build step. 

If it depends on the build having been performed then maybe you can move the clang-tidy step back to the `name: clang` for now?

Reply:
It does not depend on building the project; it only depends on the build database file generated by cmake, so running cmake beforehand is sufficient. Because cmake takes very little time, I think we can leave clang-tidy in the current place.

Reply:
Can we remove all these args except for DCMAKE_EXPORT_COMPILE_COMMANDS?  And add a comment saying why we are running cmake here.

Reply:
Should we drop `-G ninja` and also the package install above ? (Since we are not running the build it doesn't matter what build system we use I guess ?)

Reply:
Done